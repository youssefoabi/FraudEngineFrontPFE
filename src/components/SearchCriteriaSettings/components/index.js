export { default as AttributesList } from './AttributesList';
export { default as ScopesBlock } from './ScopesBlock';
export { default as FieldsBlock } from './FieldsBlock';
export { default as ParametersBlock } from './ParametersBlock';
export { default as UrlBlock } from './UrlBlock';
