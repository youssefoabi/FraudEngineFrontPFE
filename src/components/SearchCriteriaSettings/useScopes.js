import { isEmpty, isNil, map } from 'ramda';
import { useMemo } from 'react';

import useScopes from '../../screens/ScopesManagement/useScopes';

const useScopesSearchCriteriaSettings = () => {
  const { scopes } = useScopes();
  const mapScope = scope => {
    if (isNil(scope) || isEmpty(scope)) return null;

    const { name, id } = scope;

    return { value: { ...scope, scopeId: id }, label: name };
  };

  const scopesSelect = useMemo(() => map(mapScope, scopes), [scopes]);

  return { scopes: scopesSelect };
};

export default useScopesSearchCriteriaSettings;
