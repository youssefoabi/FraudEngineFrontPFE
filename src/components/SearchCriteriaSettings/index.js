import { ExpansionPanel, GridItem } from '@cdiscount/backoffice-ui';
import { func } from 'prop-types';
import { Grid } from '@material-ui/core';
import { isNil, pathOr } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';
import React from 'react';

import { enginePropTypes } from '../EngineForm/enginePropTypes';
import { FieldsBlock, ParametersBlock, ScopesBlock, UrlBlock } from './components';
import useScopes from '../../screens/ScopesManagement/useScopes';

const SearchCriteriaSettings = ({ engine, handleChange }) => {
  const { scopes: allScopes } = useScopes();

  const currentScope = pathOr(null, ['currentScope'], engine);
  const currentType1 = pathOr(null, ['inputFields', 0, 'type'], engine);
  const currentType2 = pathOr(null, ['inputFields', 1, 'type'], engine);

  if (isNilOrEmpty(allScopes)) return null;

  return (
    <ExpansionPanel title="Gestion des critÃ¨res de recherches">
      <Grid container direction="column">
        <Grid container spacing={2}>
          <GridItem item xs={12} sm={12} md={6} xl={6}>
            <FieldsBlock fieldId={0} engine={engine} handleChange={handleChange} />
          </GridItem>
          <GridItem item xs={12} sm={12} md={6} xl={6}>
            <FieldsBlock fieldId={1} engine={engine} handleChange={handleChange} />
          </GridItem>
        </Grid>
        <br />
        <ScopesBlock handleChange={handleChange} engine={engine} />
        {isNil(currentScope) ? null : (
          <>
            <UrlBlock handleChange={handleChange} engine={engine} />
            <Grid container spacing={2}>
              {isNil(currentType1) ? null : (
                <GridItem item xs={12} sm={12} md={6} xl={6}>
                  <ParametersBlock fieldId={0} engine={engine} handleChange={handleChange} />
                </GridItem>
              )}
              {isNil(currentType2) ? null : (
                <GridItem item xs={12} sm={12} md={6} xl={6}>
                  <ParametersBlock fieldId={1} engine={engine} handleChange={handleChange} />
                </GridItem>
              )}
            </Grid>
          </>
        )}
      </Grid>
    </ExpansionPanel>
  );
};

SearchCriteriaSettings.propTypes = {
  handleChange: func,
  engine: enginePropTypes,
};

SearchCriteriaSettings.defaultProps = {
  handleChange: Function.prototype,
  engine: null,
};

export default SearchCriteriaSettings;
