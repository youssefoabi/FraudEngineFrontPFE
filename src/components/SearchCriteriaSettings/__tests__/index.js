import { itRendersAllMutations } from '@cdiscount/jest-utils';

import SearchCriteriaSettings from '../';

jest.mock('react-beautiful-dnd', () => ({
  Droppable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {},
    ),
  Draggable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {},
    ),
  DragDropContext: ({ children }) => children,
}));

jest.mock('react-redux', () => ({
  useDispatch: () => ({}),
  useSelector: () => ({}),
}));

jest.resetModules();

jest.mock('../../../screens/ScopesManagement/useScopes', () => ({
  __esModule: true,
  default: jest
    .fn()
    .mockReturnValueOnce({})
    .mockReturnValueOnce({
      scopes: [],
    })
    .mockReturnValueOnce({
      scopes: null,
    })
    .mockReturnValueOnce({
      scopes: [
        {
          id: 1,
          name: 'string1',
          urlTemplate: 'stringC',
          modifiedAt: '2020-04-10T13:17:19.323Z',
        },
        {
          id: 2,
          name: 'string2',
          urlTemplate: 'stringA',
          modifiedAt: '2020-04-11T13:17:19.323Z',
        },
        {
          id: 3,
          name: 'string3',
          urlTemplate: 'stringB',
          modifiedAt: '2020-04-09T13:17:19.323Z',
        },
      ],
    }),
  namedExport: 'useScopes',
}));

const mutations = [
  { name: 'without props' },
  { name: 'with empty scopes' },
  { name: 'with null scopes' },
  {
    name: 'with scopes & props',
    props: {
      engine: {
        currentScope: { scopeId: 1 },
        inputFields: {
          0: { type: 'Address' },
          1: { type: 'Numeric' },
        },
      },
    },
  },
];

describe(SearchCriteriaSettings.name, () => {
  itRendersAllMutations(SearchCriteriaSettings, mutations);
});
