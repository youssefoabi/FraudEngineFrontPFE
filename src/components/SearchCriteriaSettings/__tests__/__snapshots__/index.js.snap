// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchCriteriaSettings SearchCriteriaSettings defaultProp functions runs defaultProps.handleChange() 1`] = `undefined`;

exports[`SearchCriteriaSettings SearchCriteriaSettings renders correctly with empty scopes 1`] = `""`;

exports[`SearchCriteriaSettings SearchCriteriaSettings renders correctly with null scopes 1`] = `""`;

exports[`SearchCriteriaSettings SearchCriteriaSettings renders correctly with scopes & props 1`] = `
<p
  defaultExpanded={false}
  title="Gestion des critÃ¨res de recherches"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    direction="column"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={2}
    >
      <h
        fullheight={false}
        item={true}
        md={6}
        sm={12}
        title={null}
        xl={6}
        xs={12}
      >
        <FieldsBlock
          engine={
            Object {
              "currentScope": Object {
                "scopeId": 1,
              },
              "inputFields": Object {
                "0": Object {
                  "type": "Address",
                },
                "1": Object {
                  "type": "Numeric",
                },
              },
            }
          }
          fieldId={0}
          handleChange={[Function]}
        />
      </h>
      <h
        fullheight={false}
        item={true}
        md={6}
        sm={12}
        title={null}
        xl={6}
        xs={12}
      >
        <FieldsBlock
          engine={
            Object {
              "currentScope": Object {
                "scopeId": 1,
              },
              "inputFields": Object {
                "0": Object {
                  "type": "Address",
                },
                "1": Object {
                  "type": "Numeric",
                },
              },
            }
          }
          fieldId={1}
          handleChange={[Function]}
        />
      </h>
    </WithStyles(ForwardRef(Grid))>
    <br />
    <ScopesBlock
      engine={
        Object {
          "currentScope": Object {
            "scopeId": 1,
          },
          "inputFields": Object {
            "0": Object {
              "type": "Address",
            },
            "1": Object {
              "type": "Numeric",
            },
          },
        }
      }
      handleChange={[Function]}
    />
    <UrlBlock
      engine={
        Object {
          "currentScope": Object {
            "scopeId": 1,
          },
          "inputFields": Object {
            "0": Object {
              "type": "Address",
            },
            "1": Object {
              "type": "Numeric",
            },
          },
        }
      }
      handleChange={[Function]}
    />
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={2}
    >
      <h
        fullheight={false}
        item={true}
        md={6}
        sm={12}
        title={null}
        xl={6}
        xs={12}
      >
        <ParametersBlock
          engine={
            Object {
              "currentScope": Object {
                "scopeId": 1,
              },
              "inputFields": Object {
                "0": Object {
                  "type": "Address",
                },
                "1": Object {
                  "type": "Numeric",
                },
              },
            }
          }
          fieldId={0}
          handleChange={[Function]}
        />
      </h>
      <h
        fullheight={false}
        item={true}
        md={6}
        sm={12}
        title={null}
        xl={6}
        xs={12}
      >
        <ParametersBlock
          engine={
            Object {
              "currentScope": Object {
                "scopeId": 1,
              },
              "inputFields": Object {
                "0": Object {
                  "type": "Address",
                },
                "1": Object {
                  "type": "Numeric",
                },
              },
            }
          }
          fieldId={1}
          handleChange={[Function]}
        />
      </h>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</p>
`;

exports[`SearchCriteriaSettings SearchCriteriaSettings renders correctly without props 1`] = `""`;
